### PR TITLE
Fixes issue where exporting check all yields a badly formatted filename. Closes #3591

### LINKED ISSUE #3591: exporting `check all` yields a badly formatted filename
Screenshot 2023-06-19 at 7 27 12 PM

### PR DESCRIPTION
**Pull request by binaek**
Screenshot 2023-06-19 at 7 29 02 PM

### CODE REVIEW

File: pkg/steampipeconfig/modconfig/parsed_name.go [hunk 1]
```diff
@@ -55,7 +55,12 @@ func (p *ParsedResourceName) ToFullNameWithMod(mod string) string {
 }
 
 func BuildFullResourceName(mod, blockType, name string) string {
```
Comment (kaidaguerre) [inline pkg/steampipeconfig/modconfig/parsed_name.go:57]: why is this being called with an empty block type?  `mod.name` is not a valid resource name. 

I think this function shgould validate it's params but the calling code should possibly not use this function

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: cmd/check.go [hunk 2]
```diff
@@ -202,6 +199,18 @@ func runCheckCmd(cmd *cobra.Command, args []string) {
 	exitCode = getExitCode(totalAlarms, totalErrors)
 }
 
+// getExportName resolves the base name of the target file
+func getExportName(targetName string, modShortName string) string {
+	parsedName, _ := modconfig.ParseResourceName(targetName)
```
Comment (kaidaguerre) [inline cmd/check.go:204]:
```
parsedName, _ := modconfig.ParseResourceName(targetName)
	if targetName == "all" {
             // there will be no block type -= manually construct name
             return fmt.Sprintf("%s.%s", modShortName, parsedName.Name)
	}
       // default to just converting to valid resource name
        return parsedName.ToFullNameWithMod(modShortName)
}

File: pkg/steampipeconfig/modconfig/parsed_name.go [hunk 3]
```diff
@@ -54,6 +54,7 @@ func (p *ParsedResourceName) ToFullNameWithMod(mod string) string {
 	return BuildFullResourceName(mod, p.ItemType, p.Name)
 }
 
+// BuildFullResourceName generates a Fully Qualified name from the given components
```
Comment (kaidaguerre) [inline pkg/steampipeconfig/modconfig/parsed_name.go:57]: no need for caps  -> `fully qualified`

give example

Comment (kaidaguerre) [reply pkg/steampipeconfig/modconfig/parsed_name.go:57]: either include validation here or raise issue for validation and put a TODO here referencing issue

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: pkg/steampipeconfig/modconfig/parsed_name.go [hunk 4]
```diff
@@ -11,6 +11,11 @@ type ParsedResourceName struct {
 	Name     string
 }
 
+// ParseResourceName splits up a resource name into it's components
```
Comment (kaidaguerre) [inline pkg/steampipeconfig/modconfig/parsed_name.go:14]: lovely - but I actually meant an example for BuildFullResourceName :)

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: pkg/steampipeconfig/modconfig/parsed_name.go (hunk 4 above)
Comment (binaek) [reply pkg/steampipeconfig/modconfig/parsed_name.go]: pushed

**Pull request merged**